### PR TITLE
Merge upstream changes from create-react-app 0.9.x

### PR DESCRIPTION
**Pull request by iamlacroix**
#### What

Merges the 0.9.x upstream changes into our fork. Via https://github.com/trunkclub/tcweb-build/pull/83

#### Why

Keeping up with the Kardashians.

#### How to test

```bash
yarn add @trunkclub/build@7.0.0-alpha.1
# or
npm install @trunkclub/build@7.0.0-alpha.1
```

#### Tested with the Following Apps

- [x] member_messenger_app (@iamlacroix)
- [ ] stylist_app
- [ ] member_profile_app
- [ ] lama
- [ ] onboarding_app
- [ ] account_app
- [ ] trunk_app

### CODE REVIEW

**Review by jblock — APPROVED** [review]
LGTM! I can't quite discern how this affects any changes we've made ourselves, but if it can build one of the internal apps with it, I think we're good 👍 

Bigger question: can we/should we run the CRA e2e suite on Circle?

File: packages/react-scripts/package.json [hunk 1]
```diff
@@ -31,49 +31,51 @@
...
     "cross-spawn": "4.0.2",
-    "css-loader": "0.26.0",
-    "detect-port": "1.0.1",
+    "css-loader": "0.26.1",
+    "detect-port": "1.1.1",
     "dotenv": "2.0.0",
     "elm-webpack-loader": "3.0.6",
-    "eslint": "3.8.1",
+    "eslint": "3.16.1",
+    "eslint-config-react-app": "^0.6.2",
```
Comment (sloanelybutsurely) [inline packages/react-scripts/package.json]: We use `@trunkclub/eslint-config` right? No need for this?

File: packages/react-scripts/package.json [hunk 2]
```diff
@@ -31,49 +31,51 @@
   "dependencies": {
     "@trunkclub/eslint-config": "3.2.0",
     "@trunkclub/react-dev-utils": "1.1.0",
```
Comment (sloanelybutsurely) [inline packages/react-scripts/package.json]: Will we need to update the version of `@trunkclub/react-dev-utils`?

File: packages/react-scripts/package.json [hunk 3]
```diff
@@ -31,49 +31,51 @@
...
     "node-sass": "3.11.1",
-    "object-assign": "4.1.0",
+    "object-assign": "4.1.1",
     "path-exists": "2.1.0",
-    "postcss-loader": "1.0.0",
+    "postcss-loader": "1.2.2",
     "progress-bar-webpack-plugin": "1.9.0",
     "promise": "7.1.1",
-    "react-dev-utils": "^0.4.2",
+    "react-dev-utils": "^0.5.2",
```
Comment (sloanelybutsurely) [inline packages/react-scripts/package.json]: We shouldn't need this package right?

File: packages/react-scripts/config/webpack.config.dev.js [hunk 4]
```diff
@@ -212,8 +203,17 @@ module.exports = {
...
       }
+      // ** STOP ** Are you adding a new loader?
+      // Remember to add the new extension(s) to the "url" loader exclusion list.
     ]
   },
+  // @remove-on-eject-begin
+  // Point ESLint to our predefined config.
+  eslint: {
+    configFile: path.join(__dirname, '../eslintrc'),
+    useEslintrc: false,
```
Comment (sloanelybutsurely) [inline packages/react-scripts/config/webpack.config.dev.js:214]: Is it possible to use `emitWarning: true` still? Without it, linting errors fail the dev builds entirely.
image

**Review by sloanelybutsurely — COMMENTED** [review]
Thanks for taking this on!

File: packages/react-scripts/package.json (hunk 2 above)
Comment (iamlacroix) [reply packages/react-scripts/package.json]: Yep, good catch.

File: packages/react-scripts/package.json (hunk 1 above)
Comment (iamlacroix) [reply packages/react-scripts/package.json]: Strange, I thought I removed it.

File: packages/react-scripts/package.json (hunk 3 above)
Comment (iamlacroix) [reply packages/react-scripts/package.json]: Again, strange. I'll remove it.

File: packages/react-scripts/config/webpack.config.dev.js (hunk 4 above)
Comment (iamlacroix) [reply packages/react-scripts/config/webpack.config.dev.js:214]: Sounds good.

**Comment by iamlacroix** [comment]
@jblock e2e breaks right now b/c of linting discrepancies.

**Comment by sloanelybutsurely** [comment]
Getting this when running `tcweb-build` in `account_app`. Looks like this is because there's a new file in utils which hasn't been published (you'll have to do so manually) and referenced in `@trunkclub/build`

```
account_app % NODE_ENV=production yarn run build
yarn run v0.23.3
$ tcweb-build build
module.js:471
    throw err;
    ^

Error: Cannot find module '@trunkclub/react-dev-utils/FileSizeReporter'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/zach/github/trunkclub/account_app/node_modules/@trunkclub/build/scripts/build.js:33:24)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

**Comment by iamlacroix** [comment]
@zperrault That's due to the `utils` dep not being version bumped. I'm pushing an update in a min. (worked locally b/c of lerna link)

**Comment by iamlacroix** [comment]
@zperrault @jblock Ok, just published `7.0.0-alpha.1` for you to try.

**Review by eanplatter — APPROVED** [review]
LGTM, tho I admit I didn't read all of it

**Pull request merged**